test/test_tableauauth_model.py

```diff
@@ -0,0 +1,22 @@
+import unittest
+import warnings
+import tableauserverclient as TSC
+
+
+class TableauAuthModelTests(unittest.TestCase):
+    def test_username_password_required(self):
+        with self.assertRaises(TypeError) as context_manager:
+            TSC.TableauAuth()
+
+    def test_site_arg_raises_warning(self):
+        with warnings.catch_warnings(record=True) as w:
+            # Cause all warnings to always be triggered.
+            warnings.simplefilter("always")
+
+            tableau_auth = TSC.TableauAuth('user',
```

Comment:
That's not the right kind of site id.

This is the site name in the URL just like tabcmd.